components/DroppableZone.tsx

```diff
@@ -0,0 +1,108 @@
+import { StyleSheet, css } from "aphrodite";
+import { ReactElement, useState } from "react";
+import Dropzone from "react-dropzone";
+import colors from "~/config/themes/colors";
+
+type Props = {
+  accept?: string;
+  multiple?: boolean;
+  children?: ReactElement;
+  noClick?: boolean;
+  handleFileDrop: (acceptedFiles: any[]) => void;
+};
+
+function DroppableZone({
+  accept,
+  multiple,
+  children,
+  noClick,
+  handleFileDrop,
+}: Props) {
+  const [isFileDragged, setIsFileDragged] = useState(false);
+
+  return (
+    <div
+      className={css(styles.dropzoneContainer, isFileDragged && styles.dragged)}
+    >
+      <Dropzone
+        accept={accept}
+        multiple={multiple}
+        noClick={noClick}
+        onDragEnter={(): void => setIsFileDragged(true)}
+        onDragLeave={(): void => setIsFileDragged(false)}
+        onDrop={(acceptedFiles) => {
+          setIsFileDragged(false);
+          handleFileDrop(acceptedFiles);
+        }}
+      >
+        {({ getRootProps, getInputProps }) => (
+          <section className={css(styles.fullCanvas)}>
+            <div {...getRootProps()} className={css(styles.dropzone)}>
+              <input {...getInputProps()} required={true} />
+              <>{children}</>
+            </div>
+          </section>
+        )}
+      </Dropzone>
+    </div>
+  );
+}
+
+const styles = StyleSheet.create({
+  dragged: {
+    backgroundColor: "#F7F7FB",
+    border: 0,
+    outline: `dotted ${colors.BLUE()}`,
+    zIndex: 999999,
+    // border: `1px dashed ${colors.BLUE()}`,
+    ":hover": {
+      borderStyle: "solid",
+    },
+    ":hover #browse": {
+      textDecoration: "underline",
+    },
+  },
+  dropzone: {
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    alignItems: "center",
+    width: "100%",
+    height: "100%",
+    boxSizing: "border-box",
+    padding: "20px 0px",
+    transition: "all ease-out 0.1s",
+  },
+  dropzoneContainer: {
+    boxSizing: "border-box",
+    width: "calc(100% - 240px)",
```

Comment:
Would this line prevent the component from being used elsewhere? Otherwise, seems like a pretty reusable solid component.